ui/src/notebooks/components/panel/NotebookPanel.tsx

```diff
@@ -75,20 +76,37 @@ const NotebookPanelHeader: FC<HeaderProps> = ({index, controls}) => {
 }
 
 const NotebookPanel: FC<Props> = ({index, children, controls}) => {
-  const {meta} = useContext(NotebookContext)
+  const {meta, updateMeta} = useContext(NotebookContext)
 
   const isVisible = meta[index].visible
+  const isFocused = meta[index].focus
 
   const panelClassName = classnames('notebook-panel', {
     [`notebook-panel__visible`]: isVisible,
     [`notebook-panel__hidden`]: !isVisible,
+    'notebook-panel__focus': isFocused,
   })
 
+  const updatePanelFocus = (focus: boolean): void => {
```

Comment:
can you wrap these in `useCallback` indexed on `[index]`? i dont think it'll have an immediate difference, it just saves some memory as we're stretching out functional components